lib/basic_auth.ex

```diff
@@ -1,11 +1,15 @@
 defmodule BasicAuth do
+  def init(options = {:application_config, _}) do
+    options
+  end
+
   def init(options) do
-    realm    = Keyword.fetch!(options, :realm)
-    username = Keyword.fetch!(options, :username)
-    password = Keyword.fetch!(options, :password)
-    %{realm: realm, username: username, password: password}
+    #Better to fail at compile time if keys are incorrect
+    [:password, :realm, :username] = Keyword.keys(options) |> Enum.sort
```

Comment:
I'm not sure I understand this change. The [`fetch!/2`](http://elixir-lang.org/docs/v1.0/elixir/Keyword.html#fetch!/2) calls would fail at compile time. Line 8 is really confusing because it read to me that what you are doing is `[:password, :realm, :username] = Keyword.keys(options)` and then piping that to `Enum.sort`, when actually it's actually assigning `(Keyword.keys(options) |> Enum.sort)`. Is this an idiom in elixir?